src/basis/entity.js

```diff
@@ -880,77 +843,24 @@
     return fn.apply(null, values);
   }
 
-  function arrayField(newArray, oldArray){
-    if (!Array.isArray(newArray))
-      return null;
-
-    if (!Array.isArray(oldArray) || newArray.length != oldArray.length)
-      return newArray || null;
-
-    for (var i = 0; i < newArray.length; i++)
-      if (newArray[i] !== oldArray[i])
-        return newArray;
-
-    return oldArray;
-  }
-
-  var reIsoStringSplit = /\D/;
-  var reIsoTimezoneDesignator = /(.{10,})([\-\+]\d{1,2}):?(\d{1,2})?$/;
-  var fromISOString = (function(){
-    function fastDateParse(y, m, d, h, i, s, ms){
-      var date = new Date(y, m - 1, d, h || 0, 0, s || 0, ms ? ms.substr(0, 3) : 0);
-      date.setMinutes((i || 0) - tz - date.getTimezoneOffset());
-      return date;
-    }
-
-    var tz;
-    return function(isoDateString){
-      tz = 0;
-      return fastDateParse.apply(
-        null,
-        String(isoDateString || '')
-          .replace(reIsoTimezoneDesignator, function(m, pre, h, i){
-            // designator formats:
-            //   <datetime>Z
-            //   <datetime>±hh:mm
-            //   <datetime>±hhmm
-            //   <datetime>±hh
-            // http://en.wikipedia.org/wiki/ISO_8601#Time_zone_designators
-            tz = Number(h || 0) * 60 + Number(i || 0);
-            return pre;
-          })
-          .split(reIsoStringSplit)
-      );
-    };
-  })();
-
-  function dateField(value, oldValue){
-    if (typeof value == 'string' && value)
-      return fromISOString(value);
-
-    if (typeof value == 'number' && isNaN(value) == false)
-      return new Date(value);
-
-    if (value == null)
-      return null;
-
-    if (value && value.constructor === Date)
-      return value;
-
-    /** @cut */ basis.dev.warn('basis.entity: Bad value for Date field, value ignored');
-    return oldValue || null;
-  }
-
   function addField(entityType, name, config){
     // normalize config
     if (typeof config == 'string' ||
-        Array.isArray(config) ||
-        (typeof config == 'function' && config.calc !== config))
+        Array.isArray(config))
     {
       config = {
         type: config
       };
     }
+    else if (typeof config == 'function' && config.calc !== config)
+    {
+      config = {
+        type: config
+      };
+
+      if ('DEFAULT_VALUE' in config.type)
```

Comment:
I guess it should be a common branch since use case is possible:

```js
createType('name', {
  field: {
    type: basis.type.something.default(...)
  }
})
```

Also we should decide which value has a priority – field `defValue` or type's `DEFAULT_VALUE`